backend/app/admin_handler.go

```diff
@@ -176,18 +240,17 @@ func (h *Handler) DeleteUsersHandler(c *gin.Context) {
 		return
 	}
 
-	err = h.db.DeleteUserByID(id)
+	err = h.svc.userRepo.DeleteUserByID(id)
```

Comment:
that's also a violation for instance in the service layer we can enforce that the user can't delete themselves or e.g if they were admin?